script/Deploy.s.sol

```diff
@@ -2,11 +2,40 @@
 pragma solidity >=0.8.19 <=0.9.0;
 
 import { WakuRlnV2 } from "../src/WakuRlnV2.sol";
+import { PoseidonT3 } from "poseidon-solidity/PoseidonT3.sol";
+import { LazyIMT } from "@zk-kit/imt.sol/LazyIMT.sol";
+import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
 import { BaseScript } from "./Base.s.sol";
 import { DeploymentConfig } from "./DeploymentConfig.s.sol";
 
 contract Deploy is BaseScript {
-    function run() public broadcast returns (WakuRlnV2 w) {
-        w = new WakuRlnV2(20);
+    function run() public broadcast returns (WakuRlnV2 w, address impl) {
+        impl = address(new WakuRlnV2());
+        bytes memory data = abi.encodeCall(WakuRlnV2.initialize, (msg.sender, 20));
+        address proxy = address(new ERC1967Proxy(impl, data));
+        w = WakuRlnV2(proxy);
+    }
+
+    function upgrade(address proxy, address newImpl) public {
+        vm.startBroadcast(msg.sender);
+        bytes memory data;
+        UUPSUpgradeable(proxy).upgradeToAndCall(newImpl, data);
+        vm.stopBroadcast();
+    }
```

Comment:
While you can run this function within tests directly, I think running this via `forge script` will not work.
You'll need a dedicated script for upgrades.

So I'd suggest moving this one into something like `script/DeployUpgrade.s.sol`

Reply:
addressed in e6d8631
I think we'll cut scope here and write that deploy script when the time comes.